cvprac/cvp_api.py

```diff
@@ -3204,6 +3210,449 @@ def workspace_build_status(self, workspace_id, build_id):
+                                                                      'name': 'mlaghealthcheck'},
+                                                           'name': 'Check MLAG Health'},
+                                            'leaf1': {'name': 'leaf1',
+                                                          'rows': {'values': [{'values': ['pre-mlag-check-l1']},
+                                                                              {'values': ['upgrade1']},
+                                                                              {'values': ['post-mlag-check-l1']}]}},
+                                            'post-mlag-check-l1': {'action': {'args': {'values': {'DeviceID': 'SN2'}},
+                                                                      'name': 'mlaghealthcheck'},
+                                                           'name': 'Check MLAG Health'},
+                                            'leaf2': {'name': 'leaf2',
+                                                          'rows': {'values': [{'values': ['pre-mlag-check-l2']},
+                                                                              {'values': ['upgrade2']},
+                                                                              {'values': ['post-mlag-check-l2']}]}}}}}
+                    }
+                    The above would result in the following hierarchy:
+                    root (series)
+                    |- left-leafs (series)
+                       |- leaf1 (series)
+                       |    |- pre-mlag-check-l1
+                       |    |- upgrade1
+                       |    |- post-mlag-check-l1
+                       |- leaf2 (series)
+                            |- pre-mlag-check-l1
+                            |- upgrade1
+                            |- post-mlag-check-l1
+
+            Returns:
+                response (dict): A dict that contains...
+                Ex: {'value': {'key': {'id':cc_id,
+                      'time': '...'}
+        '''
+        payload = custom_cc
+        url = '/api/resources/changecontrol/v1/ChangeControlConfig'
+        # For on-prem check the version as it is only supported from 2021.2.0+
+        if not self.clnt.is_cvaas:
+            if self.clnt.apiversion is None:
+                self.get_cvp_info()
+            if self.clnt.apiversion >= 6.0:
+                self.log.debug('v6 ' + str(url) + ' ' + str(payload))
+                response = self.clnt.post(url, data=payload)
+                return response
+            elif self.clnt.apiversion < 6.0:
+                    self.log.warning('change_control_create_with_custom_stages: Change Control Resource APIs are supported'
+                                     ' from 2021.2.0 or newer.')
+                    return None
+        else:
+            return self.clnt.post(url, data=payload)
```

Comment:
Simplify the API version check, response, and return calls for this function to match others.